langgraph/src/pregel/index.ts

```diff
@@ -484,7 +513,58 @@ async function executeTasks<RunOutput>(
   }
 }
 
-function _applyWrites(
+export function _shouldInterrupt(
+  checkpoint: Checkpoint,
+  interruptNodes: Array<string>,
+  snapshotChannels: Array<string>,
+  tasks: Array<PregelExecutableTask>
+): boolean {
+  const seen = checkpoint.versionsSeen[INTERRUPT];
+  const anySnapshotChannelUpdated = snapshotChannels.some(
+    (chan) => checkpoint.channelVersions[chan] > seen[chan]
+  );
+  const anyTaskNodeInInterruptNodes = tasks.some((task) =>
+    interruptNodes.includes(task.name)
+  );
+  return anySnapshotChannelUpdated && anyTaskNodeInInterruptNodes;
+}
+
+export function _localRead(
+  checkpoint: Checkpoint,
+  channels: Record<string, BaseChannel>,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  writes: Array<[string, any]>,
+  select: Array<string> | string,
+  fresh: boolean = false
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+): Record<string, any> | any {
+  if (fresh) {
+    const newCheckpoint = createCheckpoint(checkpoint, channels);
+
+    // create a new copy of channels
+    const newChannels = Object.entries(channels).reduce(
```

Comment:
Double check: This implementation is ported from a Python implementation that uses a context manager. What's the equivalent/preferred implementation in JS/TS?

Reply:
this is fine